header.go

```diff
@@ -136,6 +137,11 @@ func (h *ResponseHeader) SetStatusCode(statusCode int) {
 	h.statusCode = statusCode
 }
 
+// SetStatusLine sets response status line bytes.
+func (h *ResponseHeader) SetStatusLine(statusLine []byte) {
+	h.statusLine = statusLine
```

Comment:
This makes it impossible to reuse the bytes you provide as parameter to this function. It's better to use a pattern like the other functions: https://github.com/valyala/fasthttp/blob/0ceee57875aaf69f7e98bccb845e04fd4b1de559/header.go#L298